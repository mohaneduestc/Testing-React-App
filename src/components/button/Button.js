import React from 'react';
import './button.css';

const Button = ({label}) => (
    <div data-testid="button" className="button-style">
        {label}
    </div>
);

export default Button;
