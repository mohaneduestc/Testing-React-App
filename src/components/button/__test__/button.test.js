import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without creashing",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button/>, div);
    ReactDOM.unmountComponentAtNode(div)
})

it("renders button correctly", ()=>{
    const {getByTestId} = render(<Button label="click me pls"/>)
    expect(getByTestId('button')).toHaveTextContent("click me pls")
})

it("renders button correctly", ()=>{
    const {getByTestId} = render(<Button label="click"/>)
    expect(getByTestId('button')).toHaveTextContent("click")
})

it("renders snapshoots", ()=>{
    const tree = renderer.create(<Button label="save"/>).toJSON()
    expect(tree).toMatchSnapshot();
})