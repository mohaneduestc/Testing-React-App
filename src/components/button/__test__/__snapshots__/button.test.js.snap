// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshoots 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  save
</div>
`;
